crates/turbopack-ecmascript/src/references/external_module.rs

```diff
@@ -0,0 +1,226 @@
+use std::io::Write;
+
+use anyhow::{Context, Result};
+use turbo_tasks::Vc;
+use turbo_tasks_fs::{rope::RopeBuilder, FileContent, FileSystem, VirtualFileSystem};
+use turbopack_core::{
+    asset::{Asset, AssetContent},
+    chunk::{AsyncModuleInfo, ChunkItem, ChunkType, ChunkableModule, ChunkingContext},
+    ident::AssetIdent,
+    module::Module,
+    reference::ModuleReferences,
+    resolve::ExternalType,
+};
+
+use crate::{
+    chunk::{
+        EcmascriptChunkItem, EcmascriptChunkItemContent, EcmascriptChunkPlaceable,
+        EcmascriptChunkType, EcmascriptChunkingContext, EcmascriptExports,
+    },
+    references::async_module::{AsyncModule, OptionAsyncModule},
+    utils::StringifyJs,
+    EcmascriptModuleContent,
+};
+
+#[turbo_tasks::function]
+fn layer() -> Vc<String> {
+    Vc::cell("external".to_string())
+}
+
+#[turbo_tasks::value]
+pub struct ExternalModuleAsset {
+    pub request: String,
+    pub external_type: ExternalType,
+    pub import_externals: bool,
```

Comment:
Maybe these two field give too much freedom and it should be a new enum for this Module:

```
enum CachedExternalModuleType {
  CommonJs,
  EcmaScriptModuleViaImport,
  EcmaScriptModuleViaRequire,
}
```

Reply:
I guess the Url type is just treated as commonjs?